src/audio/chain_dma.c

```diff
@@ -121,9 +121,28 @@ static int chain_host_stop(struct comp_dev *dev)
 	int err;
 
 	err = dma_stop(cd->chan_host->dma->z_dev, cd->chan_host->index);
-	if (err < 0)
+	if (err < 0) {
+		if (err == -EBUSY) {
+			int retries = 100;
```

Comment:
you probably want to fix the misplaced brace at line 143, and while at it (optionally) you could also consider doing
```
for (int i = 0; i < 100 && err == -EBUSY; i++) {
    if (!i)
        comp_warn(...);
    k_busy_wait(10);
    err = dma_stop(...);
}
if (err < 0)
    return err;
```